### PR TITLE
fix(recipe): verify GLM-5.2 RP2 pretraining and packed SFT

### PR DESCRIPTION
**Pull request by yaoyu-33**
# What does this PR do ?

Adds the verified GLM-5.2 GB200 training recipes and model-card evidence, plus the packed-sequence metadata needed by the 8K and 128K functional runs. Chat-template, assistant-mask, EOS, and related cache-identity changes are isolated in follow-up #5361.

# Changelog

- replace the invalid GB200 pretraining corpus with bounded RedPajama2 and record the matching step-50 resume gate
- add verified 192-GB200 pretraining, 8K packed Tulu SFT, 128K long-context SFT, PEFT, and resume recipe/card coverage
- expose logical `cu_seqlens_q` / `cu_seqlens_kv` and optional physical `*_padded` boundaries for aligned packed sequences
- add an explicit maximum single-sequence length to offline packing and include it in the packed-artifact fingerprint
- add deterministic full-length 128K support data and a DSA-valid explicit PP6 layout
- retain the existing H100 verification evidence and recalculate all recorded H100/GB200 training TFLOPS from measured step times using the corrected DSA accounting from #5324
- add focused recipe, packing, builder, and training-step tests

# GitHub Actions CI

Validation completed:

- `uvx pre-commit run --all-files` on the final PR1 tree
- model-card schema validation
- focused packing-builder, packed-dataset, recipe, and conversation-processing coverage on the pre-split superset
- corrected 192-GPU RP2 pretrain and step-50 resume completed with finite losses and zero skipped/NaN iterations
- corrected 192-GPU 8K packed Tulu SFT completed 100 steps with finite loss, zero skipped/NaN iterations, and a complete step-100 checkpoint
- corrected 192-GPU 128K packed SFT completed 20 contiguous optimizer steps with finite loss and zero skipped/NaN iterations; the final 10 steps averaged 143,248.820 ms and 88.848 TFLOPS/GPU
- all recorded training TFLOPS values were re-derived from existing average step times; no training workload was rerun

The deterministic synthetic cohort verifies 128K execution support; the pinned-Tulu 8K run supplies real-data convergence evidence. The standard `uv run` environment is unavailable on macOS because `nvidia-resiliency-ext` ships Linux-only wheels, so Linux CI is requested below.

# Before your PR is "Ready for review"

- [x] Make sure you read and followed Contributor guidelines
- [x] Did you write any new necessary tests?
- [x] Did you add or update any necessary documentation?
- [x] Does the PR affect components that are optional to install? No

# Additional Information

- Follow-up template/EOS PR: #5361
- Related to MB-1140

### CODE REVIEW

File: examples/model_verification_cards/glm5-2/card.yaml [hunk 1]
```diff
@@ -258,31 +265,43 @@ items:
...
+      bridge_commit: cc190939aad0e1ee0edfb98c5455ae3fe833140b  # pragma: allowlist secret
       enabled_features:
         sequence_packing: offline
+        context_parallel_size: 4
+        moe_dispatcher: hybridep
       command: >
         ./scripts/training/train.sh --nodes 48 --gpus-per-node 4
-        --recipe glm52_sft_192gpu_gb200_bf16_config --mode sft
-        --pretrained_checkpoint work/model-verification/glm5-2/gpu-megatron/iter_0000000
+        --recipe glm52_sft_config --mode sft
```
[the remark this answers is not in the document]
Comment (yaoyu-33) [reply examples/model_verification_cards/glm5-2/card.yaml]: Fixed: the card now uses glm52_sft_192gpu_gb200_bf16_config, which is the exported GB200 8K SFT recipe.

File: examples/model_verification_cards/glm5-2/card.yaml [hunk 2]
```diff
@@ -342,43 +361,47 @@ items:
...
+        moe_dispatcher: hybridep
       command: >
         ./scripts/training/train.sh --nodes 48 --gpus-per-node 4
-        --recipe glm52_sft_192gpu_gb200_bf16_128k_config --mode sft
-        --pretrained_checkpoint work/model-verification/glm5-2/gpu-megatron/iter_0000000
-        --max_steps 20 --seq_length 131072 --lr 1e-6 --min_lr 0 --warmup_iters 2
-        tokenizer.chat_template_path=work/templates/glm52_chat_template_generation.jinja
-        checkpoint.load=null ddp.check_for_nan_in_grad=true
-        ddp.check_for_large_grads=true rerun_state_machine.check_for_nan_in_loss=true
+        --recipe glm52_sft_128k_config --mode sft
```
[the remark this answers is not in the document]
Comment (yaoyu-33) [reply examples/model_verification_cards/glm5-2/card.yaml]: Fixed: the card now uses glm52_sft_192gpu_gb200_bf16_128k_config, which is the exported GB200 128K SFT recipe.

File: examples/model_verification_cards/glm5-2/card.yaml [hunk 3]
```diff
@@ -429,7 +452,7 @@ items:
         initial_loss: 1.831100
         final_loss: 0.8957523
         last_10_steps_step_time_ms_avg: 68588.270
-        last_10_steps_model_tflops_per_gpu_avg: 9.330
+        last_10_steps_model_tflops_per_gpu_avg: 1.423
```
[the remark this answers is not in the document]
Comment (yaoyu-33) [reply examples/model_verification_cards/glm5-2/card.yaml:457]: Rechecked against the corrected DSA FLOPs accounting; 1.423 is intentional. The H100 and GB200 PEFT runs have the same model, sequence length, GBS, and LoRA configuration, so the per-step work is shared: 19.423 TFLOPS/GPU × 4.63855 s × 208 GPUs = 18,739.6678 TFLOP/step. Applying the GB200 measured time gives 18,739.6678 ÷ (68.58827 s × 192 GPUs) = 1.423019 TFLOPS/GPU, recorded as 1.423.
